modules/fedimint-lnv2-tests/src/bin/self_payment_with_one_gateway.rs

```diff
@@ -0,0 +1,96 @@
+use devimint::version_constants::VERSION_0_4_0_ALPHA;
+use devimint::{cmd, util};
+use fedimint_core::core::OperationId;
+use fedimint_lnv2_client::{FinalReceiveState, FinalSendState};
+use lightning_invoice::Bolt11Invoice;
+use substring::Substring;
+use tracing::info;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    devimint::run_devfed_test(|dev_fed| async move {
+        let fedimint_cli_version = util::FedimintCli::version_or_default().await;
+        let fedimintd_version = util::FedimintdCmd::version_or_default().await;
+        let gatewayd_version = util::Gatewayd::version_or_default().await;
+
+        if fedimint_cli_version < *VERSION_0_4_0_ALPHA {
+            info!(%fedimint_cli_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        if fedimintd_version < *VERSION_0_4_0_ALPHA {
+            info!(%fedimintd_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        if gatewayd_version < *VERSION_0_4_0_ALPHA {
+            info!(%gatewayd_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        let fed = dev_fed.fed().await?;
+
+        let gw_lnd = dev_fed.gw_lnd().await?;
```

Comment:
Can we write these tests that is agnostic to the type of gateway? The v1 tests I'm doing do this, so that we can also test the CLN gateway, but this will also be useful once we have LDK.